### PR TITLE
[hwinfo] add new port

### PR DESCRIPTION
**Pull request by Hlongyu**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Hlongyu** [comment]
> @Hlongyu please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.请阅读以下贡献者许可协议（CLA）。如果您同意 CLA，请回复以下信息。
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options: 选项：
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.（默认 - 未指定公司）我对我提交的内容拥有知识产权的唯一所有权，并且我不会在为我的雇主工作的过程中提交内容。
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.（当公司提供时）我在为我的雇主工作的过程中提交意见（或者我的雇主根据合同或适用法律对我提交的意见拥有知识产权）。我已获得雇主的许可，可以代表我的雇主提交意见并签订本协议。通过在下面签名，定义的术语“您”包括我和我的雇主。
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement
> 贡献者许可协议
> ### Contribution License Agreement
> 贡献许可协议
> This Contribution License Agreement (**“Agreement”**) is agreed to by the party signing below (**“You”**),本贡献许可协议（ **“协议”** ）由以下签署方（ **“您”** ）同意， and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your并向 M​​icrosoft Corporation 及其附属公司（“Microsoft”）转让某些许可权利，以供您使用 contributions to Microsoft open source projects. This Agreement is effective as of the latest signature对 Microsoft 开源项目的贡献。本协议自最新签署之日起生效 date below. 日期如下。
> 
> 1. **Definitions**.**定义**。
>    **“Code”** means the computer software code, whether in human-readable or machine-executable form,**“代码”**是指计算机软件代码，无论是人类可读的还是机器可执行的形式，
>    that is delivered by You to Microsoft under this Agreement.您根据本协议向 Microsoft 交付的信息。
>    **“Project”** means any of the projects owned or managed by Microsoft and offered under a license**“项目”**是指 Microsoft 拥有或管理并根据许可提供的任何项目
>    approved by the Open Source Initiative ([www.opensource.org](http://www.opensource.org)).经开源倡议 ( [www.opensource.org](http://www.opensource.org) ) 批准。
>    **“Submit”** is the act of uploading, submitting, transmitting, or distributing code or other content to any**“提交”**是指向任何人上传、提交、传输或分发代码或其他内容的行为
>    Project, including but not limited to communication on electronic mailing lists, source code control项目，包括但不限于电子邮件列表上的通信、源代码控制
>    systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of由项目管理或代表项目管理的系统和问题跟踪系统，目的是
>    discussing and improving that Project, but excluding communication that is conspicuously marked or讨论和改进该项目，但排除明显标记或
>    otherwise designated in writing by You as “Not a Submission.”您以其他方式书面指定为“非提交内容”。
>    **“Submission”** means the Code and any other copyrightable material Submitted by You, including any**“提交内容”**是指您提交的代码和任何其他受版权保护的材料，包括任何
>    associated comments and documentation.相关评论和文档。
> 2. **Your Submission**. You must agree to the terms of this Agreement before making a Submission to any**您的提交**。在提交任何内容之前，您必须同意本协议的条款
>    Project. This Agreement covers any and all Submissions that You, now or in the future (except as项目。本协议涵盖您现在或将来提交的任何及所有提交内容（除非
>    described in Section 4 below), Submit to any Project.如下第 4 节所述），提交至任何项目。
> 3. **Originality of Work**. You represent that each of Your Submissions is entirely Your original work.**作品的原创性**。您声明您提交的每份内容完全是您的原创作品。
>    Should You wish to Submit materials that are not Your original work, You may Submit them separately如果您希望提交非您原创作品的材料，您可以单独提交
>    to the Project if You (a) retain all copyright and license information that was in the materials as You如果您 (a) 保留材料中的所有版权和许可信息，则可以访问该项目
>    received them, (b) in the description accompanying Your Submission, include the phrase “Submission收到它们，(b) 在您提交的内容随附的描述中，包括短语“提交
>    containing materials of a third party:” followed by the names of the third party and any licenses or other包含第三方的材料：”后跟第三方的名称和任何许可证或其他内容
>    restrictions of which You are aware, and (c) follow any other instructions in the Project’s written您了解的限制，以及 (c) 遵循项目书面中的任何其他说明
>    guidelines concerning Submissions.有关提交的指南。
> 4. **Your Employer**. References to “employer” in this Agreement include Your employer or anyone else**你的雇主**。本协议中提及的“雇主”包括您的雇主或任何其他人
>    for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your您提交信息时代表谁，例如作为承包商、供应商或代理人。如果你的
>    Submission is made in the course of Your work for an employer or Your employer has intellectual提交是在您为雇主工作的过程中进行的，或者您的雇主有智力
>    property rights in Your Submission by contract or applicable law, You must secure permission from Your根据合同或适用法律，您提交的内容拥有产权，您必须获得您的许可
>    employer to make the Submission before signing this Agreement. In that case, the term “You” in this雇主在签署本协议之前提交该意见。在这种情况下，本文中的“您”一词
>    Agreement will refer to You and the employer collectively. If You change employers in the future and协议将指您和雇主的统称。如果您将来更换雇主并且
>    desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement希望为新雇主提交额外的意见，则您同意签署新协议
>    and secure permission from the new employer before Submitting those Submissions.并在提交这些意见之前获得新雇主的许可。
> 5. **Licenses**.**许可证**。
> 
> * **Copyright License**. You grant Microsoft, and those who receive the Submission directly or**版权许可**。您授予 Microsoft 以及那些直接收到提交内容的人或
>   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the间接来自 Microsoft，是一项永久的、全球性的、非独占的、免版税的、不可撤销的许可
>   Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute提交复制、准备衍生作品、公开展示、公开表演和分发
>   the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third提交内容和此类衍生作品，并将上述任何或所有权利再许可给第三方
>   parties. 派对。
> * **Patent License**. You grant Microsoft, and those who receive the Submission directly or**专利许可**。您授予 Microsoft 以及那些直接收到提交内容的人或
>   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under间接来自 Microsoft，永久的、全球性的、非独占的、免版税的、不可撤销的许可
>   Your patent claims that are necessarily infringed by the Submission or the combination of the您的专利权利要求必然受到提交内容或提交内容的组合的侵犯
>   Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and提交其所提交的项目以制造、已经制造、使用、要约出售、出售和
>   import or otherwise dispose of the Submission alone or with the Project.单独或与项目一起进口或以其他方式处置提交材料。
> * **Other Rights Reserved**. Each party reserves all rights not expressly granted in this Agreement.**保留其他权利**。各方保留本协议中未明确授予的所有权利。
>   No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are没有任何额外的许可或权利（包括但不限于任何默示许可）
>   granted by implication, exhaustion, estoppel or otherwise.通过暗示、用尽、禁止反言或其他方式授予。
> 
> 6. **Representations and Warranties**. You represent that You are legally entitled to grant the above**陈述和保证**。您声明您有合法权利授予上述权利
>    licenses. You represent that each of Your Submissions is entirely Your original work (except as You may许可证。您声明您提交的每份内容完全是您的原创作品（除非您可能
>    have disclosed under Section 3). You represent that You have secured permission from Your employer to已根据第 3 条披露）。您声明您已获得雇主的许可
>    make the Submission in cases where Your Submission is made in the course of Your work for Your如果您的提交是在您为您工作的过程中提交的，则提交该提交
>    employer or Your employer has intellectual property rights in Your Submission by contract or applicable雇主或您的雇主根据合同或适用的规定对您提交的内容拥有知识产权
>    law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You法律。如果您代表您的雇主签署本协议，则您声明并保证您
>    have the necessary authority to bind the listed employer to the obligations contained in this Agreement.拥有必要的权力来约束所列雇主遵守本协议中包含的义务。
>    You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS除非您选择这样做，否则您不需要为您提交的内容提供支持。除非
>    REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES适用法律要求或书面同意，保证除外
>    EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS第 3、4 和 6 节明确规定，根据本协议提供的提交内容是
>    PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF不提供任何形式的保证，包括但不限于任何保证
>    NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.非侵权、适销性或特定用途的适用性。
> 7. **Notice to Microsoft**. You agree to notify Microsoft in writing of any facts or circumstances of which**通知微软**.您同意以书面形式通知 Microsoft 任何事实或情况
>    You later become aware that would make Your representations in this Agreement inaccurate in any您后来意识到，这将使您在本协议中的陈述在任何方面都不准确
>    respect. 尊重。
> 8. **Information about Submissions**. You agree that contributions to Projects and information about**有关提交的信息**。您同意对项目的贡献和信息
>    contributions may be maintained indefinitely and disclosed publicly, including Your name and other贡献可以无限期保留并公开披露，包括您的姓名和其他信息
>    information that You submit with Your Submission.您随提交内容一起提交的信息。
> 9. **Governing Law/Jurisdiction**. This Agreement is governed by the laws of the State of Washington, and**适用法律/司法管辖区**。本协议受华盛顿州法律管辖，并且
>    the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,双方同意位于金县的联邦法院拥有专属管辖权和审判地，
>    Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to华盛顿州，除非不存在联邦主题管辖权，在这种情况下，双方同意
>    exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all华盛顿州金县高级法院拥有专属管辖权和审判地。当事人全部放弃
>    defenses of lack of personal jurisdiction and forum non-conveniens.以缺乏属人管辖权和法院不方便为由进行抗辩。
> 10. **Entire Agreement/Assignment**. This Agreement is the entire agreement between the parties, and**整个协议/转让**。本协议是双方之间的完整协议，并且
>     supersedes any and all prior agreements, understandings or communications, written or oral, between取代双方之间任何及所有先前的书面或口头协议、谅解或沟通
>     the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.与本协议主题相关的各方。本协议可由 Microsoft 转让。

@microsoft-github-policy-service agree [company="{your company}"]

**Comment by Hlongyu** [comment]
@microsoft-github-policy-service agree

**Comment by Hlongyu** [comment]
> @Hlongyu the command you issued was incorrect. Please try again.
> 
> Examples are:
> 
> ```
> @microsoft-github-policy-service agree
> ```
> 
> and
> 
> ```
> @microsoft-github-policy-service agree company="your company"
> ```

@microsoft-github-policy-service agree

**Comment by Hlongyu** [comment]
@jimwang118 Why x86_windows failed? I dont know what happend with ci log.

File: ports/hwinfo/usage [hunk 1]
```diff
@@ -0,0 +1,4 @@
+hwinfo provides CMake targets :
+    # this is heuristically generated, and may not be correct
+    find_package(hwinfo CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE hwinfo::hwinfo)
```
Comment (jimwang118) [inline ports/hwinfo/usage:4]:
```suggestion
    target_link_libraries(main PRIVATE hwinfo::hwinfo)
 
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Review by jimwang118 — DISMISSED** [review]

**Comment by jimwang118** [comment]
Usage test pass with following triplet:

```
x64-windows
```

**Comment by moritz-h** [comment]
I think this is the wrong library using the name hwinfo. I think here happened the same as in #35465 which was then reverted in #35993. Therefore may quote @dg0yt here:

> @vicroms IMO this port must be immediately renamed to lfreist-hwinfo.
> 
> > * [x]   The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
> 
> I think this is wrong. All repology references point to https://github.com/opensuse/hwinfo: https://repology.org/project/hwinfo/information

**Comment by Hlongyu** [comment]
Sorry, i search hwinfo in Browse packages and there is not port named hwinfo, so i add it.  Should i change name in this pr?

**Comment by jimwang118** [comment]
> Sorry, i search hwinfo in Browse packages and there is not port named hwinfo, so i add it. Should i change name in this pr?

You open a new PR and change the port name.

**Pull request merged**